### ISSUE TITLE
bugfixing request

### ISSUE THREAD

**Issue by Battlepixel** [tsunamayo/Starship-EVO#4269]
### Short Description of the request:
I would be very happy if you answered the three requests with a short _yes_ (when?) or _no / not in the foreseeable future)_

1. In the past I started a little video project that I haven't been able to continue with since August, because of two bugs. Would you please fix the following two bugs?
- turrets not working #4224 #4221 #4134 
- gimble targeting not working #4124 #4156 #4134 

2. Is it possible to make logic-triggerd guns also gimble? I tried to work around the broken gimble with selfmade gattlings on fighters, but they also didnt work with ai because of the mentioned bug. Try the primary weapons of the two fighters, they only fire as long as you hold down the mousebutton. In my opinon this accurate fast lasershots are looking very cool and less like sawn shotgung fire. But prior to the bug, logic triggerd weapons did not work with gimble. Another thing with the gattling barrels is, that I never got them to dont doing selfdamage, because of their low projectile speed and poor accuracy

[Walküre HF.zip](https://github.com/tsunamayo/Starship-EVO/files/7583963/Walkure.HF.zip)
[Defender HF.zip](https://github.com/tsunamayo/Starship-EVO/files/7583964/Defender.HF.zip)

To reproduce the Turret-Bug(s) try this BP:
[Bismarck.zip](https://github.com/tsunamayo/Starship-EVO/files/7588970/Bismarck.zip)

3. Would you take the time to bring the multiplayer back to a state where at least building together with friends is possible again?

**Comment by ProPeach** [comment]
There was a question regarding multiplayer in the recent Q&A that you might find useful, timestamped at 37:11 - https://youtu.be/tdjH_gyQnjc?t=2231

**Comment by Battlepixel** [comment]
That answers (3.). Thx

**Comment by Michael-Phoenix** [comment]
1. is still true for 21w47b
![image](https://user-images.githubusercontent.com/43548840/143305057-7d1b0b5d-4a3a-4ca7-ad7f-401b9aff4299.png)

**Comment by tsunamayo** [comment]
hi @Battlepixel
So 1) should come very shortly. I will do a revamp of Hotkey soon - and I will fix the gimbal.
2) I am not sure for this one. gimbal is a bit tricky to do, I will think about a way to do it neatly. How about this: You would have to still link the hotkey to your laser so the laser understand it needs to gimbal, but I would add an option so that it actually dont fire so you can control when it fire.
3) well that one might take a bit more time. I also want to change the network layer to use Steam one as people are making a huge fuss of having to open to port forward and are leaving bad review because of that...
What issue did you have regarding simply building?

Cheers

**Comment by Asnro** [comment]
> 2) I am not sure for this one. gimbal is a bit tricky to do, I will think about a way to do it neatly. How about this: You would have to still link the hotkey to your laser so the laser understand it needs to gimbal, but I would add an option so that it actually dont fire so you can control when it fire.

This solution sounds perfect to me! Having both gimbal data and a separate firing trigger would make sequenced/staggered weapon systems vastly better, as currently they can only fire straight forward with no gimbal.

**Comment by Battlepixel** [comment]
I'm really excited to hear that fixes for "1)" & "2)" are coming soon

> 2) I am not sure for this one. gimbal is a bit tricky to do, I will think about a way to do it neatly. How about this: You would have to still link the hotkey to your laser so the laser understand it needs to gimbal, but I would add an option so that it actually dont fire so you can control when it fire.
I agree to Asnro. This solution sounds simple and would give us all Freedom to set up rates of fire and also modes of fire.

> What issue did you have regarding simply building?
->3) We tried it but did not find a work around: As for simply building in multiplayer. We tried it frequently and many times.
- _Mainproblem:_ the client could never change or use any objects. The client couldn't flip switches, place anything, use seats, etc. 
_The following alone wouldn't matter, but it may have something to do with the main problem:_
- Usually the client and host saw vehicles in slightly to completely different positions. 
- The client only saw buildings that were spawned before joining. (That alone wouldn't matter, but it may have something to do with the main problem.)
